internal/engine/synclet_build_and_deployer.go

```diff
@@ -87,6 +87,22 @@ func (sbd *SyncletBuildAndDeployer) deployInfoForImageBlocking(ctx context.Conte
 	return deployInfo, ok
 }
 
+func (sbd *SyncletBuildAndDeployer) forgetImage(ctx context.Context, img reference.NamedTagged) error {
+	sbd.deployInfoMu.Lock()
+	defer sbd.deployInfoMu.Unlock()
+
+	imgNameAndTag := docker.ToImgNameAndTag(img)
+
+	deployInfo, ok := sbd.deployInfo[imgNameAndTag]
```

Comment:
are we concerned about the cases where an image is deployed more than once, to different pods?

Reply:
I think we should be, but that's also broken independently of this change (i.e., we're already keying podid + synclet client by image)

Reply:
👀 
uhhh.
well the answer to your question is "no". but maybe that's not the RIGHT answer...